CHANGELOG

```diff
@@ -1,6 +1,125 @@
 devel
 -----
 
+* ArangoDB now has an optional AQL query execution plan cache that can be used
+  to skip query planning and optimization when running the same queries 
+  repeatedly. This feature is currently experimental.
+
+  The plan cache is fully optional and is only considered for AQL queries that
+  have the option `optimizePlanForCaching` set to `true`. If the option is not
+  set to `true`, the plan cache is bypassed. This is also the default.
+
+  If the option is set to `true`, the query's eligibility for being cached is
+  checked first. Any queries that have any of the following query options set
+  are not eligible for plan caching:
+  - `allPlans`
+  - `explainRegisters`
+  - `forceOneShardAttributeValue`
```

Comment:
Wasn't there an customer explicitly requiring this feature?

Reply:
Fixed in CHANGELOG.